NewForce-Cohort-6/tabloidmvc-the-caramel-apple-pops
# Ja ticket12 editacategory

Trello Ticket: [#12 - Edit A Category](https://trello.com/c/AB14rRIr/12-12-edit-a-category)
Created code to allow a logged in user to Edit a Category.  
Created the following files and methods:
Added Edit button in view/category/index.cshtml
Added GET/POST Edit methods in CategoryController
Added Update Category in CategoryRepository
Added Update Category in ICategoryRepository
Added Edit.cshtml file in view/category

Type of change
 New feature (non-breaking change which adds functionality)
  
Testing Instructions

- git add and git commit whatever you're working on on your branch
- git checkout main
- git fetch --a
- git checkout ja-ticket12-editacategory
- If you don't see the most recent version of my files, git pull origin ja-ticket12-editacategory
- If you haven't created the Tabloid MVC database, please do so.
- Use this to fill out tables and columns: [SQL data](https://github.com/NewForce-Cohort-6/tabloidmvc-the-caramel-apple-pops/blob/main/SQL/01_TabloidMVC_Create_DB.sql)
- Use this to seed the database: [SQL Seed Data](https://github.com/NewForce-Cohort-6/tabloidmvc-the-caramel-apple-pops/blob/main/SQL/02_TabloidMVC_Seed_Data.sql)

- Click the execute with debugger solid green button to launch the app
- In browser, please go to Login on the nav bar and log in with this email: admin@example.com
- This should redirect you back to the home page.
- Click on Category Management in NavBar
- View the list of available Categories (categories should appear in alphabetical order)

(List of Categories should include the list below):
CloseMagic
Cooking
Cthulhu Sightings
Entertainment
Fun
History
Home and Garden
Humor
Improv
Movies
Music
Politics
Regrets
Science
Technology

- When a user is on the Category list page
- They select the `Edit` button that has a pencil on it beside the chosen category
- To test this the user should choose the edit button beside Technology
- Then they should be directed to a form and given the ability to change the Category’s name
- Here the user should change the category name
- When the user is finished updating the category information they click the ‘Save’ button’
- This should save the updated Category in the database to what the user changed it to.
- When they click ‘Save’ the user should be redirected to the new Category list page which should have the edited version of the category in the list.
- If the user decides not to edit the Category they should click the ‘Cancel’ button
- Then the user should be redirected back to the list page and the category they chose should remain unaltered.

 **NOTE:** For the time being it is acceptable to treat all users as `admin` users. There is a future story about enforcing user permissions.

# Checklist:
